### PR TITLE
MAINT: adapt for upstream odl API changes, closes #26

### LINKED ISSUE #26: installation with local odl
I managed to install odlcuda with odl that comes from pip. However, that version is outdated so I would like to use odlcuda with the latest odl version. Following exactly the same installation steps as with the off-the-shelf odl version, my odl cannot find "cuda" as an implementation, e.g. 

> NotImplementedError: no corresponding data space available for space FunctionSpace(IntervalProd([-333.8016, -333.8016,    0.    ], [ 333.8016 ,  333.8016 ,  257.96875]), out_dtype='float32') and implementation 'cuda'

This is all strange, because it does find my odl version within the installation and this odl version works perfectly fine without cuda. Also my system finds odlcuda as it shows up on auto-complete after `import odl`. Any ideas what might have gotten wrong here? What is the mechanism that tells odl that odlcuda is present?

### PR DESCRIPTION
**Pull request by kohr-h**

### CODE REVIEW

File: odlcuda/cu_ntuples.py [hunk 1]
```diff
@@ -195,494 +322,265 @@ def element(self, inp=None, data_ptr=None):
...
-    impl = 'cuda'
+        Calculate ``z = a * x + b * y`` using optimized CUDA routines.
 
-    @staticmethod
-    def available_dtypes():
-        """Return the available data types."""
-        return CUDA_DTYPES
+        Parameters
+        ----------
+        a, b : `LinearSpace.field` `element`
```
Comment (adler-j) [inline odlcuda/cu_ntuples.py]: Is \`field.element\` not ok here?

File: odlcuda/cu_ntuples.py [hunk 2]
```diff
@@ -195,494 +322,265 @@ def element(self, inp=None, data_ptr=None):
...
-        if field is None or field == RealNumbers():
-            return np.dtype('float32')
-        elif field == ComplexNumbers():
-            return NotImplemented
-        else:
-            raise ValueError('no default data type defined for field {}'
-                             ''.format(field))
+        return self.weighting.inner(x1, x2)
 
+    def _integral(self, x):
```
Comment (adler-j) [inline odlcuda/cu_ntuples.py:378]: Since when is this in?

**Review by adler-j — APPROVED** [review]
Very minor stuff

File: odlcuda/cu_ntuples.py (hunk 2 above)
Comment (kohr-h) [reply odlcuda/cu_ntuples.py:378]: Has always been in the cuda backend :-)

File: odlcuda/cu_ntuples.py (hunk 1 above)
Comment (kohr-h) [reply odlcuda/cu_ntuples.py]: No idea. Docs are not built anyway, so I can write anything.

**Comment by kohr-h** [comment]
All fixed, I also removed the free weighting functions, like in the CPU spaces.

**Pull request merged**